thenewboston_node/business_logic/storages/file_system.py

```diff
@@ -185,5 +187,5 @@ def _write_file(self, file_path: Path, binary_data: bytes, mode):
         if self._is_finalized(file_path):
             raise exceptions.FinalizedFileWriteError(f'Could not write to finalized file: {file_path}')
 
-        with file_path.open(mode=mode) as fo:
+        with atomic_write_append(file_path, mode=mode, dir=tempfile.gettempdir()) as fo:
```

Comment:
I am worried about filename collisions here. Is it possible?

I was thinking that we would utilize either https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory or https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile .

Also why do not we encapsulate dealing with temporary directories into atomic_write_append() ?

Reply:
`tempfile` library is used under the hood of `atomicwrites` so we shouldn't worry about filename collisions